samples/csharp_dotnetcore/language-generation/05.a.multi-turn-prompt-with-language-fallback/Dialogs/UserProfileDialog.cs

```diff
@@ -116,7 +116,7 @@ private async Task<DialogTurnResult> ConfirmStepAsync(WaterfallStepContext stepC
             }, stepContext.Context.Activity.Locale));
 
             // We can send messages to the user at any point in the WaterfallStep.
-            await stepContext.Context.SendActivityAsync(ActivityFactory.FromObject(msg.ToString()), cancellationToken);
+            await stepContext.Context.SendActivityAsync(ActivityFactory.FromObject(msg.Text.ToString()), cancellationToken);
```

Comment:
Something still looks off here.  There is a call to ActivityFactory.FromObject above, online 113.  The 2nd call to ActivityFactory seems redundant.  

Should this just be sending out the msg as it is?

Reply:
You are right. I hadn't looked at some of the surrounding code. This happens in both instances. I tested and sending just the constructed `msg` is enough.